### ISSUE TITLE
StompConfig Dynamic configuration ！

### ISSUE THREAD

**Issue by MicroLord** [stomp-js/ng2-stompjs#32]
how to custom settings the stompConfig's header , get the token when  user login in , i want to transform the token to the stompConfig's header , How to implement ？ thanks

**Comment by kum-deepak** [comment]
We also face the same issue. Unfortunately I am unable to find a satisfactory solution so far. It seems to be limitation of the way Angular works.

Currently we are using little bit convoluted way as a workaround. Basically our Login page is not part of the Angular App, so that when Angular app initializes we already have the token which can be used by other services.

Alternatively, you can create a 'Holder' class and inject that as a service. Instantiate and put the StompService into this when you have the token with you.

**Comment by MicroLord** [comment]
thank u for your idea， and could you show some demos , like 'Holder' class , or links , thanks again !

**Comment by kum-deepak** [comment]
Give me some time, I will write a sample outlining the approach.

**Comment by ShinDarth** [comment]
I'm experiencing the same issue

**Comment by ShinDarth** [comment]
I'm experiencing the same issue. It's not possible for us to use this library at all because of this limitation :(

**Comment by jwz104** [comment]
I have found an easy way, but I think it only works when the web sockets are used in parts of the application where a token is provided, so when a user isn't logged in the websockets won't work.  
This is my configuration:
```
export const stompConfig: StompConfig = {
  url: '...',
  headers: {
    Authorization: localStorage.getItem('token'),
  },
  heartbeat_in: 0,
  heartbeat_out: 20000,
  reconnect_delay: 5000,
  debug: true,
};
```
I am lucky that my application always requires an user to be logged in(except for registration/login).

But I agree that there should be a way to dynamically configure the stomp service.

The problem is that the stomp service works via decency injection and makes a connection when it is injected, so if you want to configure it in the constructor you are already to late. There is probably a way to solve this.

Maybe the websocket connection should not be established in the `StompService` constructor, but manually started.

**Comment by kum-deepak** [comment]
Please check branch `dyna`. It has one more service `StompRService`. This does not need config in the constructor.

Inject this service into your class and to connect:

```javascript
stompRService.config = config;
stompRService.initAndConnect();
```

If you are sharing this service across your application the above code need to be called only once.

You do not need to check if stomp has been initialized before making calls to subscribe and publish.
If underlying service is yet to initialize, these will get queued up and execute when underlying stomp service is connected. This should simplify other calls.

`initAndConnect` can actually be called multiple times (in case you wish to force reconnect) with potentially updated `config`.

**Comment by kum-deepak** [comment]
The changes have been moved to public released version 0.6.1. If no further issues reported this issue will be closed in few days.
